packages/solidstart/src/vite/sourceMaps.ts

```diff
@@ -1,53 +1,48 @@
 import { sentryVitePlugin } from '@sentry/vite-plugin';
-import type { Plugin } from 'vite';
+import type { Plugin, UserConfig } from 'vite';
 import type { SentrySolidStartPluginOptions } from './types';
 
 /**
- * A Sentry plugin for SolidStart to enable source maps and use
- * @sentry/vite-plugin to automatically upload source maps to Sentry.
- * @param {SourceMapsOptions} options
+ * A Sentry plugin for adding the @sentry/vite-plugin to automatically upload source maps to Sentry.
  */
-export function makeSourceMapsVitePlugin(options: SentrySolidStartPluginOptions): Plugin[] {
+export function makeAddSentryVitePlugin(options: SentrySolidStartPluginOptions, viteConfig: UserConfig): Plugin[] {
   const { authToken, debug, org, project, sourceMapsUploadOptions } = options;
-  return [
-    {
-      name: 'sentry-solidstart-source-maps',
-      apply: 'build',
-      enforce: 'post',
-      config(config) {
-        // TODO(v9): Remove this warning
-        if (config.build?.sourcemap === false) {
-          // eslint-disable-next-line no-console
-          console.warn(
-            "[Sentry SolidStart Plugin] You disabled sourcemaps with the `build.sourcemap` option. Currently, the Sentry SDK will override this option to generate sourcemaps. In future versions, the Sentry SDK will not override the `build.sourcemap` option if you explicitly disable it. If you want to generate and upload sourcemaps please set the `build.sourcemap` option to 'hidden' or undefined.",
-          );
-        }
-
-        // TODO(v9): Remove this warning and print warning in case source map deletion is auto configured
-        if (!sourceMapsUploadOptions?.filesToDeleteAfterUpload) {
-          // eslint-disable-next-line no-console
-          console.warn(
-            "[Sentry SolidStart Plugin] The Sentry SDK has enabled source map generation for your SolidStart app. If you don't want to serve Source Maps to your users, either configure the `filesToDeleteAfterUpload` option with a glob to remove source maps after uploading them, or manually delete the source maps after the build. In future Sentry SDK versions source maps will be deleted automatically after uploading them.",
-          );
-        }
 
-        return {
-          ...config,
-          build: {
-            ...config.build,
-            sourcemap: true,
-          },
-        };
-      },
-    },
+  let updatedFilesToDeleteAfterUpload: string[] | undefined = undefined;
+
+  if (
+    typeof sourceMapsUploadOptions?.filesToDeleteAfterUpload === 'undefined' &&
+    typeof sourceMapsUploadOptions?.unstable_sentryVitePluginOptions?.sourcemaps?.filesToDeleteAfterUpload ===
+      'undefined' &&
+    // Only if source maps were previously not set, we update the "filesToDeleteAfterUpload" (as we override the setting with "hidden")
+    typeof viteConfig.build?.sourcemap === 'undefined'
+  ) {
+    // This also works for adapters, as the source maps are also copied to e.g. the .vercel folder
```

Comment:
l: any chance this comment is from astro (not sure if solid start has adapters)? :)

Reply:
oh yes, good catch